exercise1/terraform/app_gw.tf

```diff
@@ -0,0 +1,167 @@
+#___________________________ App Gateway Related resources ______________________________
+resource "azurerm_subnet" "ex1_subnet_app_gw" {
+  name                 = "${var.rg_name}_subnet_app_gw"
+  resource_group_name  = azurerm_resource_group.ex1.name
+  virtual_network_name = azurerm_virtual_network.ex1_vnet.name
+  address_prefixes     = ["10.0.3.0/24"]
+}
+
+resource "azurerm_user_assigned_identity" "ex1_app_gw_ass_iden" {
+  name                = "${var.rg_name}_app_gw_ass_iden"
+  location            = azurerm_resource_group.ex1.location
+  resource_group_name = azurerm_resource_group.ex1.name
+}
+
+resource "azurerm_network_security_group" "ex1_app_gw_netsecg" {
+  name                = "${var.rg_name}_app_gw"
+  location            = azurerm_resource_group.ex1.location
+  resource_group_name = azurerm_resource_group.ex1.name
+}
+
+resource "azurerm_subnet_network_security_group_association" "ex1_secg_asso_app_gw" {
+  subnet_id                 = azurerm_subnet.ex1_subnet_app_gw.id
+  network_security_group_id = azurerm_network_security_group.ex1_app_gw_netsecg.id
+}
+
+# NSG Rules required ports by application gateway please see https://learn.microsoft.com/en-us/azure/application-gateway/configuration-infrastructure#:~:text=V2%3A%20Ports%2065200%2D65535
+
+# Inbound Client traffic
+resource "azurerm_network_security_rule" "https_rule_app_gw" {
+  name                        = "AllowHTTPS"
+  priority                    = 100
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = "443"
+  source_address_prefixes     = [azurerm_subnet.ex1_subnet_app_gw.address_prefixes.0]
```

Comment:
This is backwards. You allow any since that is your entrypoint into your app. Destination is the GW subnet. They way you have it now you will not be able to access your app externally.

Reply:
yes, i have corrected it.